pandas/io/parsers.py

```diff
@@ -234,18 +234,28 @@ class ParserWarning(Warning):
 fields if it is not spaces (e.g., '~').
 """ % (_parser_params % _fwf_widths)
 
+def get_compression(filepath_or_buffer, encoding, compression_kwd):
+    """
+    Determine the compression type of a file or buffer.
```

Comment:
move this entire function to ``pandas.io.common``; call it maybe ``get_compression_type``.

This should be a bit more lightweight as you can see there are already other functions which *use* the compression. This will simply infer from a keywork and/or file extension and return the type of compression. Which can then be passed to other routines.